mcs/class/test-helpers/SocketResponder.cs

```diff
@@ -91,8 +91,11 @@ private void Listen ()
 					listenSocket = tcpListener.AcceptSocket ();
 					listenSocket.Send (requestHandler (listenSocket));
 					try {
-						listenSocket.Shutdown (SocketShutdown.Receive);
+						// On Windows a Receive() is needed here before Shutdown() to consume the data some tests send.
+						listenSocket.ReceiveTimeout = 1000;
```

Comment:
Only one test seems to timeout so I changed the timeout to 10s like you suggested to be on the safe side.